api/src/main/java/org/apache/apex/api/ApexPlugin.java

```diff
@@ -18,12 +18,15 @@
  */
 package org.apache.apex.api;
 
+import org.apache.hadoop.classification.InterfaceStability;
+
 import com.datatorrent.api.Component;
 
 /**
  * Marker interface for ApexPlugins.
  * @param <T>
  */
+@InterfaceStability.Evolving
 public interface ApexPlugin<T extends ApexPluginContext> extends Component<T>
```

Comment:
Plugin is too generic and will often require disambiguation (whether it is the IDE asking you which Plugin to import or whether searching for something like that in a global space like google search for example). I am not too hung up on the name and ok with Plugin.